tests/gmsh/complex/test_greet.py

```diff
@@ -0,0 +1,384 @@
+    box_fused.set_region("box")
+    # box_fused.set_mesh_step(1)
+    all_obj = [box_fused]
+
+    mesh_all = [*all_obj]
+
+    gen.make_mesh(mesh_all)
+    gen.write_mesh(mesh_name + ".msh2", gmsh.MeshFormat.msh2)
+
+
+def test_splitting():
+    """
+    In this test, we split the object (cylinder) into chosen number of parts.
+    We should provide a function for this.
+    Question is how to set the splitting plane, how to provide the axis in which we split the object.
+
+    The main point is in the end, where we transform ObjectSet into list of ObjectSet,
+    taking advantage of the simple problem.. We will have to use the symetric fragmentation and then select
+    properly the parts...
+    We should think of creating method for half-space defined by a plane..
+    """
+    mesh_name = "splitting"
+    gen = gmsh.GeometryOCC(mesh_name, verbose=True)
+
+    # tunnel_start = np.array([-50, -50, 10])
+    tunnel_start = np.array([0, 0, 0])
+    tunnel_end = np.array([50, 50, 10])
+    radius = 5
+
+    tunnel = gen.cylinder(radius, tunnel_end-tunnel_start, tunnel_start)
+
+    # cutting box
+    box_s = 50
+    # cutting plane between the cylinders
+    split_plane = gen.rectangle([box_s, box_s])
+    # normal is in z-axis
+    z = [0, 0, 1]
+
+    # directional vector of the cylinder
+    u_t = tunnel_end - tunnel_start
+    u_t = u_t / np.linalg.norm(u_t)  # normalize
+
+    # axis of rotation
+    axis = np.cross(z, u_t)
+    axis = axis / np.linalg.norm(axis)  # normalize
+
+    angle = np.arccos(np.dot(u_t, z))
+    split_plane.rotate(axis=axis, angle=angle, center=[0, 0, 0])
+
+    splits = []
+    length_t = np.linalg.norm(tunnel_end-tunnel_start)
+    n_parts = 5  # number of parts
+    length_part = length_t / n_parts  # length of a single part
+
+    split_pos = tunnel_start + length_part*u_t
+    for i in range(n_parts-1):
+        split = split_plane.copy().translate(split_pos)
+        splits.append(split)
+        split_pos = split_pos + length_part*u_t
+
+    # tunnel_f = tunnel.fragment(*splits)
+    tunnel_f = tunnel.fragment(*[s.copy() for s in splits])
+
+    # split fragmented ObjectSet into list of ObjectSets by dimtags
+    tunnel_parts = []
+    for dimtag, reg in tunnel_f.dimtagreg():
+        tunnel_parts.append(gmsh.ObjectSet(gen, [dimtag], [gmsh.Region.default_region[3]]))
+
+    def center_comparison(obj):
+        center, mass = obj.center_of_mass()
+        return np.linalg.norm(center-tunnel_start)
+
+    # for t in tunnel:
+    #     print(center_comparison(t))
+
+    tunnel_parts.sort(reverse=False, key=center_comparison)
+
+    for t in tunnel_parts:
+        print(gen.model.getMass(*(t.dim_tags[0])))
+        t.set_mesh_step(2.0)
+
+    gen.make_mesh([*tunnel_parts, *splits])
+    gen.write_mesh(mesh_name + ".msh2", gmsh.MeshFormat.msh2)
+
+
+def test_fuse_tunnel():
+    """
+    Test shows, how to fuse two cylinders cross-secting each in a common plane under given angle.
```

Comment:
So the problem is just how to make a fuse to behave nicely. Can we simplify this procedure by rotating the second tunnel around the outer extremal point of the intersection ellipse? 

Looking forward to the possible generalization of the concept.

Reply:
I suppose I have the two cylinders given by their end points and radius.
Suppose, I found the extremal point EP. It still takes two steps:

1. move the second cylinder so its "corner" aligns to the EP
(why would I need any rotation? I have the cylinder defined by its endpoints..)

2. prolong the first cylinder from the intersection center so that its "corner" is aligned with EP

I called "corner" the point at the base of the cylinder, which corresponds to the rectangle node, when looking on the plane in which the cylinders lie.


I can probably move the second cylinder so that the corners of both align. However, the intersection point and the end of the second cylinder would moved away from the given end points...

Reply:
camera:/Samsung%2520Galaxy%2520models%2520(MTP)@usb:001,010/store_00010001/DCIM/Camera/20200325_181440.jpg

Reply:
Jsem myslel ze mam delky l1 a l2 a \alpha, ale ty mas rovnou konce cylindru a tim i bod C.
Tak pak staci spocist \delta a natahout ty cylindry, aby to vypadalo jako na obrazku a dotkly se presne v E. A pak se jen udela fuse. Tim EP jsem myslel bod E.

Reply:
created test.. however unsatisfying result with cylindery edges sticking out (bad geometry, no problem in gmsh)